sdk-development-guidelines.md

```diff
@@ -0,0 +1,200 @@
+* MUST have a sample app - a minimal app that integrates the SDK and shows how to use the SDK features. It SHOULD be written in Kotlin (Android) or Swift (iOS).
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
+    * MUST use [gradle](https://docs.gradle.org/current/userguide/userguide.html) with [gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) to build Android library or Java library
+    * MUST have consistent minSdk and supportLib versions with other SDKs
+        * SHOULD use [shared build configuration](https://github.com/rakutentech/android-buildconfig) for consistent version with other SDKs/libraries
+        * MAY use manual configuration with values (same as in [shared build configuration](https://github.com/rakutentech/android-buildconfig)), if so
+            * SHOULD use minSdk 23
+            * SHOULD use latest versions of [Android X](https://developer.android.com/jetpack/androidx) libraries (if required)
+        * Example 1: using the [auto value](https://github.com/google/auto/tree/master/value) annotation processor in version 1.4-rc1 is allowed because it is not exported to SDK clients
+        * Example 2: using [okhttp](https://github.com/square/okhttp) compile/implementatino dependency in version 3.4.0-RC1 is not allowed because that dependency will be imposed onto SDK clients
+* For iOS <img src="images/ios.png" alt="drawing" width="15"/>:
+    * SHOULD be built from a [CocoaPods](https://guides.cocoapods.org/) Podspec or [SPM](https://www.swift.org/package-manager/)
+    * SHOULD use [fastlane](https://docs.fastlane.tools/) to automate tasks and MAY use our [shared build configuration](https://github.com/rakutentech/ios-buildconfig) to help with setup and maintain consistency
+    * SHOULD target iOS 12.0 and above, MAY target a more recent iOS version
+    * Xcode unit test project SHOULD pull the module's source from a local Podspec path reference
+    * Xcode sample project MAY be separate from the unit test project and sample project SHOULD pull the module's source from a local Podspec path reference
+    * Xcode project SHOULD build using the latest version of Xcode and the iOS SDK with no warnings
+
+### Programming Language
+
+#### Production Code
+
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
+    * MUST use Kotlin or Java. Kotlin is RECOMMENDED
+    * RECOMMENDED to use Java 11 language features
+    * MAY NOT use other JVM languages (except for Kotlin)
+* For iOS <img src="images/ios.png" alt="drawing" width="15"/>:
+    * MUST use Swift or Objective-C. Swift is RECOMMENDED
+        * If the SDK needs to execute code before the application is launched you SHOULD use Objective-C (e.g. `+ (void)load`) for at least that part of the code. This is because Swift [does not have](http://jordansmith.io/handling-the-deprecation-of-initialize/) reliable static initializers
```

Comment:
Change "SHOULD use Objective-C" to "MAY user Objective-C" please

Reply:
changed to suggested statement